distributed/tests/test_client.py

```diff
@@ -6515,3 +6515,30 @@ async def test_workers_collection_restriction(c, s, a, b):
     future = c.compute(da.arange(10), workers=a.address)
     await future
     assert a.data and not b.data
+
+
+@gen_cluster(client=True)
+async def test_custom_key_with_batches(c, s, a, b):
```

Comment:
Would you mind moving the contents of this test over to `test_map_batch_size` so all the checks for `client.map` + `batch_size=` are co-located?

Reply:
Done.